### PR TITLE
bug-fix/internal-issue-#744

### PR DESCRIPTION
**Pull request by gnusi**
fix for https://github.com/arangodb/backlog/issues/744

### CODE REVIEW

File: arangod/Aql/IResearchViewNode.cpp [hunk 1]
```diff
@@ -1298,8 +1298,42 @@ aql::CostEstimate IResearchViewNode::estimateCost() const {
...
+
+  if (_options.restrictSources) {
+    for (auto const cid : _options.sources) {
+      visitor(cid);
+    }
+  } else {
+    _view->visitCollections(visitor);
+  }
+
+  TRI_ASSERT(!_dependencies.empty());
```
Comment (Dronplane) [inline arangod/Aql/IResearchViewNode.cpp:1334]: This assert looks useless as check for empty dependencies is at method entry

File: arangod/Aql/IResearchViewNode.cpp [hunk 2]
```diff
@@ -1298,8 +1298,42 @@ aql::CostEstimate IResearchViewNode::estimateCost() const {
...
   }
-  // TODO: get a better guess from view
-  aql::CostEstimate estimate = _dependencies[0]->getCost();
+
+  transaction::Methods& trx = _plan->getAst()->query().trxForOptimization();
+  if (trx.status() != transaction::Status::RUNNING) {
+    return aql::CostEstimate::empty();
+  }
+
+  TRI_ASSERT(_plan && _plan->getAst());
```
Comment (Dronplane) [inline arangod/Aql/IResearchViewNode.cpp:1307]: Suggest to move assert above before trxForOptimization is read

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11527/

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11544/

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11545/

**Review by Dronplane — APPROVED** [review]
LGTM

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11561/

**Pull request merged**